### PR TITLE
Add new --osTmpDir optional argument

### LINKED ISSUE #966: Image size optimization seems to not happen in memory
Which is really sad because it generates a lot of fs access and slows down significantly the whole scraper. If it is not possible to make it in memory could we at least use `/dev/shm` instead of `/tmp`?

### PR DESCRIPTION
**Pull request by kelson42**
Fix #966

### CODE REVIEW

**Comment by rgaudin** [comment]
Looks good to me although I'm not in capacity to test.
@kelson42, there's a typo in the changelog line.

**Comment by kelson42** [comment]
@rgaudin Typo should be fixed.

**Comment by kelson42** [comment]
@ISNIT0 I thought you had made a comment on that?

**Comment by ISNIT0** [comment]
@kelson42 I had made a review but not "submitted" it, my bad. You should see it now

File: src/mwoffliner.lib.ts [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
 
+import fs, { readFileSync } from 'fs';
+import logger from './Logger';
+
+/* ***********************************/
+/* TMPDIR OVERRIDE HAS TO BE HANDLED */
+/* AT THE REALLY BEGIN               */
+/* ***********************************/
+
+process.argv.forEach(function(val, index, array) {
```
[the remark this answers is not in the document]
Comment (kelson42) [reply src/mwoffliner.lib.ts]: How to get this done without running all the imports? If I just put this piece of code using the current solution (so withing `execute()`) the TMPDIR change won't apply properly as it is important to be defined at compile time.

[the remark this answers is not in the document]
Comment (ISNIT0) [reply src/mwoffliner.lib.ts]: Then it should go in https://github.com/openzim/mwoffliner/blob/master/src/cli.ts

File: src/cli.ts [hunk 2]
```diff
@@ -27,6 +27,33 @@ const argv = yargs
...
+/* ***********************************/
+/* TMPDIR OVERRIDE HAS TO BE HANDLED */
+/* AT THE REALLY BEGIN               */
+/* ***********************************/
+
+import fs, { readFileSync } from 'fs';
+import logger from './Logger';
+
+if (argv.osTmpDir) {
+  const osTmpDir: string = <string>argv.osTmpDir;
```
Comment (ISNIT0) [inline src/cli.ts]: Should be `const osTmpDir = argv.osTmpDir as string;`

**Pull request merged**